amqpdispatcher/truly_robust_connection.py

```diff
@@ -0,0 +1,195 @@
+import asyncio
+import logging
+from asyncio import AbstractEventLoop, Future
+from typing import Callable, Set, Optional, Dict, Any, Type, Awaitable
+
+import aiormq
+from aio_pika import Connection, RobustChannel, Channel
+from aio_pika.exceptions import CONNECTION_EXCEPTIONS
+from aio_pika.tools import CallbackCollection
+from aio_pika.types import TimeoutType
+
+from amqpdispatcher.wait_group import WaitGroup
+
+logger = logging.getLogger(__name__)
+
+
+class TrulyRobustConnection(Connection):
+    """Truly Robust connection """
+
+    connection: Optional[aiormq.connection.Connection]  # type: ignore
+    consumer_completion_group: WaitGroup
+    _reconnect_attempt: int
+    _running_task: Optional[Future]
+    _on_reconnect_callbacks: CallbackCollection
+    _consumption_task: Optional[Callable[[], Awaitable[None]]]
+    __channels: Set[RobustChannel]
+
+    CHANNEL_CLASS = RobustChannel
+
+    def __init__(
+        self,
+        url: str,
+        loop: Optional[AbstractEventLoop] = None,
+        fail_fast: bool = False,
+    ) -> None:
+        super().__init__(loop=loop or asyncio.get_event_loop(), url=url)
+
+        self.fail_fast = fail_fast
+        self.consumer_completion_group = WaitGroup()
```

Comment:
Is this used at all?

Reply:
It isn't currently used, but I've left it in as a feature which potentially allows notification of when all consumers have completed their tasks.